articles/remapping.md

```diff
@@ -0,0 +1,77 @@
+### Remapping Names
+#### ROS 1
+In ROS 1 [Names](http://wiki.ros.org/Names) can be remapped to other names by passing [remapping arguments](http://wiki.ros.org/Remapping%20Arguments) via the command line. `roscpp` additionally allows remaps to be passed via Init as strings in STL containers. A ROS1 remap consists of two names: one that should be replaced with another.
+
+In ROS1 remapping works with FQN. When a name is to be checked for remappings, it is first expanded to a FQN. Both parts of the remapping rule are expanded to a FQN. Then a name is changed to the remap's second name only if it exactly matches the first name.
+
+#### ROS 2
+In ROS2 remapping will be more powerful. There are more types of remappings that will be supported
+
+- Namespace replacement
+
+ - Replace all or part of a namespace with another namespace
+ 
+- Basename replacement
+
+ - Replace all of the basename with another basename
+ 
+- Arbitrary substring replacement
+
+ - Replace any substring with another string
+ 
+- Exact replacement (This is ROS 1)
+
+ - Replace a name with another name exactly
+
+ROS 2 may support remapping rules being applied either before or after the names are expanded to FQN. ROS 1 only supports the latter. 
```

Comment:
You're right here as well. I guess the rationale for "pre-FQN" replace is weaker now. In the next paragraph, however, you point out that it might help with more narrowly specifying what to remap.